tests/suites/os/tests/connectivity/index.js

```diff
@@ -0,0 +1,184 @@
+      tests: ["wired", "wireless"].map((adaptor) => {
+        return {
+          title: `${adaptor.charAt(0).toUpperCase()}${adaptor.slice(1)} test`,
+          os: {
+            type: "object",
+            required: ["network"],
+            properties: {
+              network: {
+                type: "object",
+                required: [adaptor],
+                properties: {
+                  [adaptor]: {
+                    type: "boolean",
+                    const: true,
+                  },
+                },
+              },
+            },
+          },
+          run: async function (test) {
+            let testIface = "";
+            if (adaptor === "wireless") {
+              testIface = "wifi";
+            } else {
+              testIface = "ethernet";
+            }
+            const iface = await this.context
+              .get()
+              .worker.executeCommandInHostOS(
+                `nmcli d  | grep ' ${testIface} ' | grep connected | awk '{print $1}'`,
+                this.context.get().link
+              );
+
+            if (iface === "") {
+              throw new Error(`No ${testIface} interface found.`);
+            }
+
+            await this.context
+              .get()
+              .worker.executeCommandInHostOS(
+                `ping -c 10 -I ${iface} ${URL_TEST}`,
+                this.context.get().link
+              );
+
+            test.ok(true, `${URL_TEST} responded over ${testIface}`);
+          },
+        };
+      }),
+    },
+    {
+      title: "Proxy tests",
+      tests: ["redsocks", "http-connect"].map((proxy) => {
+        return {
+          title: `${proxy.charAt(0).toUpperCase()}${proxy.slice(1)} test`,
+          run: async function (test) {
+            const PROXY_PORT = 8123;
+
+            await this.context
+              .get()
+              .worker.executeCommandInHostOS(
+                "mkdir -p /mnt/boot/system-proxy",
+                this.context.get().link
+              );
+
+            const proxyState = await this.context.get().worker.proxy({
+              port: PROXY_PORT,
+            });
+
+            await this.context
+              .get()
+              .worker.executeCommandInHostOS(
+                'printf "' +
+                  "base { \n" +
+                  "log_debug = off; \n" +
+                  "log_info = on; \n" +
+                  "log = stderr; \n" +
+                  "daemon = off; \n" +
+                  "redirector = iptables; \n" +
+                  "} \n" +
+                  "redsocks { \n" +
+                  `type = ${proxy}; \n` +
+                  `ip = ${proxyState.ip}; \n` +
+                  `port = ${PROXY_PORT}; \n` +
+                  "local_ip = 127.0.0.1; \n" +
+                  "local_port = 12345; \n" +
+                  '} \n" > /mnt/boot/system-proxy/redsocks.conf',
+                this.context.get().link
+              );
+
+            // Start reboot check
```

Comment:
I was thinking we were putting down a helper for reboot, no?

Reply:
We haven't added a reboot helper into leviathan yet.